### PR TITLE
Fix the document widget order so defaults are always before other widgets.

### LINKED ISSUE #521: Make the default markdown editor the editor, not the renderer
I'm not sure how to do this, actually. I think it makes sense to make the default renderer for markdown the editor, since likely people will be writing markdown files more than viewing them. However, our defaults system automatically puts things registered for .md files above the global editor default.

Thoughts?

### PR DESCRIPTION
**Pull request by jasongrout**
Fixes #521.

### CODE REVIEW

File: src/docregistry/registry.ts [hunk 1]
```diff
@@ -215,30 +215,34 @@ class DocumentRegistry implements IDisposable {
...
-        this._widgetFactoryExtensions[ext].forEach(n => { factories.add(n); });
-      }
     }
 
     // Add the global default factory.
     if (this._defaultWidgetFactory) {
       factories.add(this._defaultWidgetFactory);
     }
 
+      // Add the extension-specific factories in registration order.
```
Comment (blink1073) [inline src/docregistry/registry.ts]: Alignment

Comment (jasongrout) [inline src/docregistry/registry.ts]: You sound like a used car shop :). Fixed.

Comment (blink1073) [inline src/docregistry/registry.ts]: dodgy-mechanic

**Comment by blink1073** [comment]
👍

**Pull request merged**